### PR TITLE
MULE-11616: allow extracting the oauth token from json

### PR DESCRIPTION
**Pull request by alepulver**

### CODE REVIEW

File: extensions/oauth/src/test/java/org/mule/test/oauth2/AbstractOAuthAuthorizationTestCase.java [hunk 1]
```diff
@@ -138,7 +136,8 @@ protected void configureWireMockToExpectTokenPathRequestForClientCredentialsGran
   protected void configureWireMockToExpectTokenPathRequestForClientCredentialsGrantType(String accessToken) {
     wireMockRule
         .stubFor(post(urlEqualTo(TOKEN_PATH)).willReturn(aResponse().withBody("{" + "\"" + ACCESS_TOKEN_PARAMETER
-            + "\":\"" + accessToken + "\"," + "\"" + EXPIRES_IN_PARAMETER + "\":\"" + EXPIRES_IN + "\"}")));
+            + "\":\"" + accessToken + "\"," + "\"" + EXPIRES_IN_PARAMETER + "\":\"" + EXPIRES_IN + "\"}")
```
Comment (elrodro83) [inline extensions/oauth/src/test/java/org/mule/test/oauth2/AbstractOAuthAuthorizationTestCase.java]: we should keep some testing that returns the data as string

File: tests/infrastructure/src/main/java/org/mule/test/infrastructure/process/rules/MuleDeployment.java [hunk 2]
```diff
@@ -80,7 +80,7 @@
   private static final boolean STOP_ON_EXIT = parseBoolean(getProperty("mule.test.stopOnExit", "true"));
   private static final boolean DEBUGGING_ENABLED = parseBoolean(getProperty("mule.test.debug", "false"));
   private static final String REMOTE_REPOSITORIES = getProperty("mule.test.repositories", "");
-  private static final String DEBUG_PORT = "5005";
+  private static final int DEBUG_PORT = 5005;
```
Comment (elrodro83) [inline tests/infrastructure/src/main/java/org/mule/test/infrastructure/process/rules/MuleDeployment.java]: changes in this class seem unrelated

File: extensions/oauth/src/test/java/org/mule/test/oauth2/internal/authorizationcode/functional/AuthorizationCodeAuthorizationCustomTokenExprTestCase.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import org.mule.extension.oauth2.internal.tokenmanager.TokenManagerConfig;
+import org.mule.runtime.api.metadata.MediaType;
+import org.mule.runtime.oauth.api.state.ResourceOwnerOAuthContext;
+import org.mule.tck.junit4.rule.DynamicPort;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * This test checks that we can extract tokens from a text response (when the server omits a mime-type or uses text/plain),
```
Comment (elrodro83) [inline extensions/oauth/src/test/java/org/mule/test/oauth2/internal/authorizationcode/functional/AuthorizationCodeAuthorizationCustomTokenExprTestCase.java]: add this as an allure documentation

**Comment by alepulver** [comment]
jenkins validate
